frontend/src/metabase/components/Header.jsx

```diff
@@ -39,104 +82,133 @@ export default class Header extends Component {
     }
 
     render() {
-        var titleAndDescription;
-        if (this.props.isEditingInfo) {
-            titleAndDescription = (
-                <div className="Header-title flex flex-column flex-full bordered rounded my1">
-                    <Input className="AdminInput text-bold border-bottom rounded-top h3" type="text" value={this.props.item.name || ""} onChange={this.setItemAttribute.bind(this, "name")}/>
-                    <Input className="AdminInput rounded-bottom h4" type="text" value={this.props.item.description || ""} onChange={this.setItemAttribute.bind(this, "description")} placeholder="No description yet" />
-                </div>
-            );
-        } else {
-            if (this.props.item && this.props.item.id != null) {
-                titleAndDescription = (
-                    <TitleAndDescription
-                        title={this.props.item.name}
-                        description={this.props.item.description}
-                    />
-                );
-            } else {
-                titleAndDescription = (
-                    <TitleAndDescription
-                        title={`New ${this.props.objectType}`}
-                        description={this.props.item.description}
-                    />
-                );
-            }
-        }
-
-        var attribution;
-        if (this.props.item && this.props.item.creator) {
-            attribution = (
-                <div className="Header-attribution">
-                    Asked by {this.props.item.creator.common_name}
-                </div>
-            );
-        }
-
-        var headerButtons = this.props.headerButtons.map((section, sectionIndex) => {
-            return section && section.length > 0 && (
-                <span key={sectionIndex} className="Header-buttonSection flex align-center">
-                    {section.map((button, buttonIndex) =>
-                        <span key={buttonIndex} className="Header-button">
-                            {button}
-                        </span>
-                    )}
-                </span>
-            );
-        });
+        const {
+            item,
+            isEditing,
+            isEditingInfo,
+            editingTitle,
+            editingSubtitle,
+            defaultActions,
+            editingActions,
+            fullscreenActions,
+            objectType,
+            isFullscreen,
+            editBarButtons
+        } = this.props
 
         return (
             <div>
-                {this.renderEditHeader()}
-                {this.renderHeaderModal()}
-                <div className={"QueryBuilder-section flex align-center " + this.props.headerClassName} ref="header">
+                { isEditing && (
+                    <EditBar
+                        title={editingTitle}
+                        subtitle={editingSubtitle}
+                        // TODO - @kdoh - shouldnt bars have standard buttons?
```

Comment:
yeah I think that Cancel/Save is a pretty standard structure.

Reply:
(at least that's the only use case for EditBar atm; only other place where it is used is the data reference section)